src/Storages/MergeTree/MergeTreeSettings.cpp

```diff
@@ -92,7 +92,7 @@ namespace ErrorCodes
     M(String, merge_workload, "", "Name of workload to be used to access resources for merges", 0) \
     M(String, mutation_workload, "", "Name of workload to be used to access resources for mutations", 0) \
     M(Milliseconds, background_task_preferred_step_execution_time_ms, 50, "Target time to execution of one step of merge or mutation. Can be exceeded if one step takes longer time", 0) \
-    \
+    M(Bool, check_table_structure_completely, false, "If true, tables must have identical definitions, including projections and secondary indices. Otherwise, the source table's projections and secondary indices must be a subset of those in the target table.", 0) \
```

Comment:
Could you please clarify the documentation? Which tables need to have identical definitions to each other? Does "Otherwise" refer to the default behavior? And is this behavior the same as the current implementation?

Reply:
Any suggestions? 
If source B and target A have the same structure, projections and secondary indices can be defined in one of four ways:
1. The projections and secondary indices of source table B are subsets of target table A
2. The projections and secondary indices of source table B are the same as that of target table A
3. The projections and secondary indices of source table B intersect with that of target table A
4. The projections and secondary indices of target table A are subsets of source table B

When `ATTACH` source table B to target table A, only 2 works correctly in the current implementation(issue [70602](https://github.com/ClickHouse/ClickHouse/issues/70602) provides a simple example), and both 1 and 2 worked correctly in previous versions(such as 23.3 which we used to use).

Reply:
fixed